subiquity/server/server.py

```diff
@@ -552,8 +553,93 @@ async def apply_autoinstall_config(self, context):
             await controller.apply_autoinstall_config()
             await controller.configured()
 
+    def _strip_controller_keys(
```

Comment:
I kind of want you to change how this function works, to have it return two dicts.
One dict of the known-good keys, one of the known-bad keys.
I think it would simplify the later operations, as
* you won't have to separately calculate bad_keys (this will tell you)
* the calculation of the good keys will be in the same spot (right now there are a few similar-looking things and the logic is spread over this function and the caller), and the calling function can just take action